File: abc279/d/main.py
import math
def f(n):
  return a/math.sqrt(n+1) + b*n


a,b=map(int,input().split())
l,r = 0,a/b

while(r - l > 10):
  n1 = (2*r+l)//3
  n2 = (r+2*l)//3
  if f(n1) > f(n2):
    l = n1
  else:
    r = n2

ans = f(l)

for i in range(l+1,r+1):
  ans = min(ans,f(i))

print(ans)

File: abc279/d/main.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,a) for(int i=0;i<a;i++)
#define rrep(i,a,b) for(int i=a;i<=b;i++)
#define drep(i,a,b) for(int i=a;i>=b;i--)
#define fore(i,a) for(auto &i:a)
#define all(x) x.begin(), x.end()
using ll = long long;
// const ll INF = 1e18;
using P = pair<int, int>;

ll a, b;

int main() {
    cin >> a >> b;
    ll l = 0, r = 1e18;
    auto f = [&](ll x) {
        double res = (double) a / sqrt(x+1) + (double) b*x;
        return res;
    };
    while(r - l > 10) {
        ll n1 = (2*l + r) / 3;
        ll n2 = (l + 2*r) / 3;
        if(f(n1) > f(n2)) l = n1;
        else r = n2;
    }
    double ans = a;
    for(ll i = l; i <= r; i++) {
        ans = min(ans,f(i));
    }
    printf("%.10lf\n", ans);
}
